nixos/modules/system/boot/loader/generic-extlinux-compatible/extlinux-conf-builder.sh

```diff
@@ -37,7 +37,9 @@ done
 
 [ "$timeout" = "" -o "$default" = "" ] && usage
 
-mkdir -p $target/nixos
+if [[ -n "@copyKernels@" ]]; then
```

Comment:
Are you sure `copyKernels = true` causes `@copyKernels` to be empty and `copyKernels = false` causes it to be empty?

Pretty sure the boolean value is passed as string, so your check does not make sense to me.
See
https://github.com/NixOS/nixpkgs/blob/5823018b1b27b9675a51a84a7fc9cdd44327fa3e/nixos/modules/system/boot/loader/grub/install-grub.pl#L71

Reply:
I was using [generations-dir-builder.sh](https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/system/boot/loader/generations-dir/generations-dir-builder.sh) and [raspberrypi-builder.sh](https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/system/boot/loader/raspberrypi/raspberrypi-builder.sh) as references.

Bools are passed into the environment (and thus to substituteAll) as the empty string when false, and the string `1` when true. I didn't know this before but I verified it manually.

The grub module does a crazy thing where it passes the config through an xml file, so that's not relevant here (look at how `get` is defined)

Reply:
Actually I just realized there's a problem _here_: this mkdir is necessary even if copyKernels = false, if it's on a different fs. Any objections to just leaving this mkdir in place all the time? It'll leave an empty dir if not copying, but otherwise harmless.